src/main.rs

```diff
@@ -1,14 +1,71 @@
 #![feature(proc_macro_hygiene, decl_macro)]
 extern crate rocket_contrib;
+extern crate ws;
+
+mod piece_state;
+
+use crate::piece_state::PieceState;
+
+use std::cell::Cell;
+use std::rc::Rc;
 
 use rocket_contrib::serve::StaticFiles;
 use std::io;
+use ws::{listen, CloseCode, Handler, Handshake, Message, Result, Sender};
+
+use serde::{Deserialize, Serialize};
+
+struct Server {
+    out: Sender,
+    count: Rc<Cell<u32>>,
+}
+
+impl Handler for Server {
+    fn on_open(&mut self, shake: Handshake) -> Result<()> {
+        return Ok(self.count.set(self.count.get() + 1));
+    }
+
+    fn on_message(&mut self, msg: Message) -> Result<()> {
+        // parse the msg as text
+        if let Ok(text) = msg.into_text() {
```

Comment:
The way I understand this, the if statement will execute if `msg.into_text()` is an enum like `Ok(text)`. I think you should handle the possible error case instead of just returning `Ok(())`.